src/Credentials/CredentialProvider.php

```diff
@@ -692,6 +706,35 @@ private static function loadDefaultProfiles() {
         return $profiles;
     }
 
+    public static function getAssumeRoleCredentials(
+        $sourceProfileName = '',
+        $profileName = '',
+        $filename = '',
+        $config = []
+    )
+    {
+        $data = self::loadProfiles($filename);
+        $credentialSource = !empty($data[$profileName]['credential_source']) ? $data[$profileName]['credential_source'] : null;
+        if (isset($credentialSource) && is_string($credentialSource)) {
+            switch ($credentialSource) {
+                case 'Environment':
+                    return self::env();
```

Comment:
For each of the cases, if, say in this case, the env() doesn't contain proper credentials, does it make sense to instead throw a new intermediate exception explicitly specifying that the env credentials were specified by the credential source and not found? I'm thinking of the scenario in which a customer may see an env exception, but not realize why it's failing there (instead of going through the full default chain).

Reply:
Whats the most idiomatic way of throwing this exception?  Since self::env returns a callback, i can't put a try catch around this specifically.  The only thing I could think off off the top of my head is to add an optional parameter like so:

    public static function env($isCredentialsSource = false){
//happy path here, removed for brevity
            $errorMessage = 'Could not find environment variable '
                . 'credentials in ' . self::ENV_KEY . '/' . self::ENV_SECRET);
            if ($isCredentialsSource){
                $errorMessage = "Provided credential_source 'Environment' in credentials failed to retrieve credentials" .
                                              "with exception: " . $errorMessage
             }
            return self::reject();
        };
}

//please ignore the awful spacing I wrote this on github